lib/widgets/message_list.dart

```diff
@@ -1104,10 +1107,15 @@ class StreamMessageRecipientHeader extends StatelessWidget {
               store.topicVisibilityPolicy(message.streamId, topic))),
         ]));
 
+    // When already in a topic narrow, disable tap interaction that would just
+    // push a MessageListPage for the same topic narrow.
+    // TODO(#1039) simplify by removing topic-narrow condition if we remove
+    //   recipient headers in topic narrows
     return GestureDetector(
-      onTap: () => Navigator.push(context,
-        MessageListPage.buildRoute(context: context,
-          narrow: TopicNarrow.ofMessage(message))),
+      onTap: !inTopicNarrow
+        ? () => Navigator.push(context, MessageListPage.buildRoute(context: context,
+            narrow: TopicNarrow.ofMessage(message)))
```

Comment:
nit: avoid reformatting code you aren't touching; so leave `MessageListPage.buildRoute` on a new line after `Navigator.push`